src/main/java/com/googleintern/wfm/ruleengine/action/GeneralRulesForWorkgroupId.java

```diff
@@ -0,0 +1,76 @@
+package src.main.java.com.googleintern.wfm.ruleengine.action;
+
+import com.google.common.collect.*;
+import src.main.java.com.googleintern.wfm.ruleengine.model.FilterModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.PoolAssignmentModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.RuleModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.UserPoolAssignmentModel;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * GeneralRulesForWorkgroupId class is used to create rules that can apply to all users from the
+ * same work group.
+ */
+public class GeneralRulesForWorkgroupId {
+  /** Generate rules that can apply to all users from the same workgroup Id. */
+  public static ImmutableSet<RuleModel> generalRuleByWorkgroupId(
+      ImmutableListMultimap<Long, UserPoolAssignmentModel> userPoolAssignmentsByWorkGroupIdBuilder,
+      Long workgroupId) {
+    ImmutableList<UserPoolAssignmentModel> userPoolAssignmentsFromSameWorkGroupId =
+        userPoolAssignmentsByWorkGroupIdBuilder.get(workgroupId);
+    ImmutableSet<PoolAssignmentModel> permissionIntersections =
+        findPermissionIntersection(userPoolAssignmentsFromSameWorkGroupId);
+    ImmutableSetMultimap<Long, Long> permissionGroup =
+        groupPermissionByCasePoolId(permissionIntersections);
+    Long workforceId = userPoolAssignmentsFromSameWorkGroupId.get(0).workforceId();
+    return createGeneralRuleForWorkgroupId(permissionGroup, workforceId, workgroupId);
+  }
+
+  private static ImmutableSet<PoolAssignmentModel> findPermissionIntersection(
+      List<UserPoolAssignmentModel> userPoolAssignmentsFromSameWorkGroupId) {
+    ImmutableSet<PoolAssignmentModel> permissionIntersections =
+        userPoolAssignmentsFromSameWorkGroupId.get(0).poolAssignments();
+
+    for (UserPoolAssignmentModel user : userPoolAssignmentsFromSameWorkGroupId) {
+      permissionIntersections =
+          ImmutableSet.copyOf(Sets.intersection(permissionIntersections, user.poolAssignments()));
+      if (permissionIntersections.size() == 0) {
+        return ImmutableSet.of();
+      }
+    }
+    return permissionIntersections;
```

Comment:
I think that permissionIntersections can just be a Set here, and you can return ImmutableSet.copyOf(permissionIntersections) at the end. 
Otherwise, the permissionIntersections ImmutableSets will keep getting created only to be thrown away.